gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/ManifestBasedDataset.java

```diff
@@ -132,9 +132,10 @@ public Iterator<FileSet<CopyEntity>> getFileSetIterator(FileSystem targetFs, Cop
             copyableFile.setFsDatasets(srcFs, targetFs);
             copyEntities.add(copyableFile);
 
-            Path fromPath = srcFs.getFileStatus(fileToCopy).isDirectory() ? fileToCopy : fileToCopy.getParent();
+            // Always grab the parent since the above permission setting should be setting the permission for a folder itself
```

Comment:
can you add to the comment more details why we need/can skip the last dir, and also comment about these permissions being set in other place.